satellite/peer.go

```diff
@@ -491,6 +491,21 @@ func ignoreCancel(err error) error {
 	return err
 }
 
+func (peer *Peer) restartLoop(ctx context.Context, name string, service func(context.Context) error) error {
+	for {
```

Comment:
PS: this needs to handle context cancellation properly

Reply:
does this mean call `ignoreCancel` inside here?

Reply:
egon's saying we need to see if the ctx that's passed in is still valid before continuing through the for loop. if the context is canceled, we should stop looping